RUFAS/graph_generator.py

```diff
@@ -156,13 +158,56 @@ def generate_graph(
                 graph_details, filter_file_name, graphics_dir
             )
 
-            return log_pool
+            return all_logs
         except Exception as e:
             raise e
 
+    def _validate_graph_filter(self, graph_details: Dict[str, str | List[str]]
+                               ) -> List[Dict[str, str] | Dict[str, Any]]:
+        """
+        Ensures all the filter keys are valid and if not, raises an error and reports them back to Output Manager.
+
+        Parameters
+        ----------
+        graph_details : Dict[str, str | List[str]]
+            A dictionary containing details/metadata about the graph.
+
+        Returns
+        -------
+        List[Dict[str, str] | Dict[str, Any]]
+            The logs, warnings, and errors to be reported to OutputManager.
+
+        """
+        required_graph_filter_keys = ["type", "filters", "variables"]
+        optional_graph_filter_keys = list(FIGURE_SETTERS.keys()) + list(AXES_SETTERS.keys())
```

Comment:
```suggestion
        required_graph_filter_keys = ["type", "filters"]
        optional_graph_filter_keys = list(FIGURE_SETTERS.keys()) + list(AXES_SETTERS.keys() + ["variables"])
```
As I understand it, "variables" is only needed when one of the variables filtered out by the filter is a dictionary and you want to pull values out of that dictionary.

Reply:
Right. That makes sense and we do have a check for those variables being present when the filter is a dictionary (and that would be logged to the OM). I can make this change. Thanks, Ed!